### PR TITLE
feat: new bullet component

### PR DESCRIPTION
**Pull request by Exayy**
## Type

- Feature

### Summarise concisely:

Add a new component : Bullet

### CODE REVIEW

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Bullet/index.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  tooltipBaseId,
+}) => (
+  <Tooltip baseId={tooltipBaseId} text={tooltip}>
+    <StyledContainer variant={variant} size={size} className={className}>
+      {icon ? <Icon name={icon} size={iconSizes[size]} /> : text}
+    </StyledContainer>
+  </Tooltip>
+)
+
+const customXorPropType = (
```
Comment (chambo-e) [inline src/components/Bullet/index.tsx:105]: What if we put icon and text as optional ?

File: src/components/Bullet/index.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  variant = 'default',
+  size = 'medium',
+  icon,
+  text,
+  tooltip,
+  tooltipBaseId,
+}) => (
+  <Tooltip baseId={tooltipBaseId} text={tooltip}>
+    <StyledContainer variant={variant} size={size} className={className}>
+      {icon ? <Icon name={icon} size={iconSizes[size]} /> : text}
```
Comment (chambo-e) [inline src/components/Bullet/index.tsx:100]: Can we use percentage as Icon size ?

File: src/components/Bullet/index.tsx (hunk 1 above)
Comment (Exayy) [reply src/components/Bullet/index.tsx:105]: it works fine but you have no Proptypes warning if both aren't specified

File: src/components/Bullet/__stories__/index.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    storyDescription: 'Add a `tooltip` using tooltip property',
+  },
+}
+Tooltip.decorators = [
+  () => (
+    <div style={{ display: 'flex' }}>
+      <Bullet text="1" tooltip="tooltip text" />
+    </div>
+  ),
+]
```
Comment (matthprost) [inline src/components/Bullet/__stories__/index.stories.tsx:99]:
```suggestion
export const Tooltip = Template.bind({})
Tooltip.parameters = {
  docs: {
    storyDescription: 'Add a `tooltip` using tooltip property',
  },
}
Tooltip.args = {
  text: '1',
  tooltip: 'tooltip text'
}
```

Comment (matthprost) [inline src/components/Bullet/__stories__/index.stories.tsx:99]:
```suggestion
export const Tooltip = Template.bind({})
Tooltip.parameters = {
  docs: {
    storyDescription: 'Add a `tooltip` using tooltip property',
  },
}
Tooltip.args = {
  text: '1',
  tooltip: 'tooltip text'
}
```

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Bullet/index.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    width: 32px;
+    height: 32px;
+    font-size: 16px;
+  `,
+  small: css`
+    width: 24px;
+    height: 24px;
+    font-size: 14px;
+  `,
+} as const
```
Comment (matthprost) [inline src/components/Bullet/index.tsx:54]:
```suggestion
const sizes = {
  medium: `
    width: 32px;
    height: 32px;
    font-size: 16px;
  `,
  small: `
    width: 24px;
    height: 24px;
    font-size: 14px;
  `,
} as const
```

doesn't it work this way ?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Bullet/__stories__/index.stories.tsx (hunk 3 above)
Comment (Exayy) [reply src/components/Bullet/__stories__/index.stories.tsx:99]: without a Flexbox around tooltip, the tooltip position is bad (Tooltip component issue not related with Bullet)

File: src/components/Bullet/index.tsx (hunk 1 above)
Comment (chambo-e) [reply src/components/Bullet/index.tsx:100]: Yeah, don't care about proptypes :D
Less logic we add the better

**Review by matthprost — APPROVED** [review]

**Comment by matthprost** [comment]
All good to me, well done 🎉

**Review by vincentaudebert — APPROVED** [review]

**Pull request merged**